From 030482d6d03510ea73ec7f8fcd4aaf36c37721dd Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Wed, 7 Jun 2017 19:50:22 +0200
Subject: [PATCH] Don't search for NetworkManager or ConnMan anymore

GNetworkMonitor abstracts this away and the dependencies are unused
since PackageKit was ported.
---
 configure.ac    | 39 ---------------------------------------
 src/Makefile.am |  8 --------
 2 files changed, 47 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9c63db85b..2d36e26e1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -168,7 +168,6 @@ dnl - Library dependencies
 dnl ---------------------------------------------------------------------------
 GLIB_REQUIRED=2.46.0
 GIO_REQUIRED=2.16.1
-NETWORK_MANAGER_REQUIRED=0.6.4
 POLKIT_GOBJECT_REQUIRED=0.98
 
 dnl ---------------------------------------------------------------------------
@@ -201,49 +200,12 @@ AC_CHECK_HEADERS([execinfo.h])
 AC_CHECK_FUNCS(clearenv)
 AC_PATH_PROG(GMSGFMT, msgfmt, msgfmt)
 
-# unix networking stack always available
-networking_apis="unix"
-
 dnl ---------------------------------------------------------------------------
 dnl - Haiku compatibility
 dnl ---------------------------------------------------------------------------
 AC_CHECK_FUNCS(setpriority)
 
 dnl ---------------------------------------------------------------------------
-dnl - NetworkManager (default enabled)
-dnl ---------------------------------------------------------------------------
-AC_ARG_ENABLE(networkmanager, AS_HELP_STRING([--disable-networkmanager],[Disable NetworkManager support]),
-		enable_networkmanager=$enableval)
-if test x$enable_networkmanager != xno; then
-	PKG_CHECK_MODULES(NETWORK_MANAGER, NetworkManager >= $NETWORK_MANAGER_REQUIRED, HAVE_NETWORKMANAGER="yes", HAVE_NETWORKMANAGER="no")
-	if test "x$HAVE_NETWORKMANAGER" = "xyes"; then
-		networking_apis="${networking_apis},NetworkManager"
-		AC_DEFINE(HAVE_NETWORKMANAGER, 1, [define if NetworkManager is installed])
-	else
-		if test x$enable_networkmanager = xyes; then
-			AC_MSG_ERROR([NetworkManager enabled but not found])
-		fi
-	fi
-fi
-
-dnl ---------------------------------------------------------------------------
-dnl - ConnMan (default disabled)
-dnl ---------------------------------------------------------------------------
-AC_ARG_ENABLE(connman, AS_HELP_STRING([--enable-connman],[Enable ConnMan support]),
-	      enable_connman=$enableval, enable_connman=no)
-if test x$enable_connman != xno; then
-	PKG_CHECK_MODULES(CONNMAN, connman, HAVE_CONNMAN="yes", HAVE_CONNMAN="no")
-	if test "x$HAVE_CONNMAN" = "xyes"; then
-		networking_apis="${networking_apis},ConnMan"
-		AC_DEFINE(HAVE_CONNMAN, 1, [define if Connection Manager is installed])
-	else
-		if test x$enable_connman = xyes; then
-			AC_MSG_ERROR([Connman enabled but not found])
-		fi
-	fi
-fi
-
-dnl ---------------------------------------------------------------------------
 dnl - Use systemd and logind rather than ConsoleKit
 dnl ---------------------------------------------------------------------------
 AC_ARG_ENABLE(systemd, AS_HELP_STRING([--enable-systemd],[enable systemd and logind code]),
@@ -631,7 +593,6 @@ echo "
         Able to run locally:       ${enable_local}
         Use systemd:               ${enable_systemd}
         Enable offline update:     ${enable_offline_update}
-        Networking stacks:         ${networking_apis}
         GStreamer plugin:          ${build_gstreamer_plugin}
         Pango module (GTK+3):      ${build_gtk_module}
         gobject-introspection:     ${found_introspection}
diff --git a/src/Makefile.am b/src/Makefile.am
index f3d476804..830aaf873 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -9,8 +9,6 @@ AM_CPPFLAGS =						\
 	$(SQLITE_CFLAGS)				\
 	$(POLKIT_CFLAGS)				\
 	$(PYTHON_CFLAGS)				\
-	$(NETWORK_MANAGER_CFLAGS)			\
-	$(CONNMAN_CFLAGS)				\
 	$(SYSTEMD_CFLAGS)				\
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"		\
 	-DBINDIR=\"$(bindir)\"				\
@@ -97,8 +95,6 @@ packagekit_direct_CPPFLAGS =				\
 packagekit_direct_LDADD =				\
 	$(GLIB_LIBS)					\
 	$(GMODULE_LIBS)					\
-	$(NETWORK_MANAGER_LIBS)				\
-	$(CONNMAN_LIBS)					\
 	$(SQLITE_LIBS)					\
 	$(PK_GLIB2_LIBS)				\
 	$(POLKIT_LIBS)					\
@@ -126,8 +122,6 @@ packagekitd_CPPFLAGS =					\
 packagekitd_LDADD =					\
 	$(GLIB_LIBS)					\
 	$(GMODULE_LIBS)					\
-	$(NETWORK_MANAGER_LIBS)				\
-	$(CONNMAN_LIBS)					\
 	$(SQLITE_LIBS)					\
 	$(PK_GLIB2_LIBS)				\
 	$(PYTHON_LIBS)					\
@@ -155,8 +149,6 @@ pk_self_test_LDADD =					\
 	$(GLIB_LIBS)					\
 	$(GMODULE_LIBS)					\
 	$(SQLITE_LIBS)					\
-	$(NETWORK_MANAGER_LIBS)				\
-	$(CONNMAN_LIBS)					\
 	$(PK_GLIB2_LIBS)				\
 	$(POLKIT_LIBS)					\
 	$(GIO_LIBS)					\
